engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -460,6 +460,234 @@ static int GraphicsTextureTypeToImageType(int texturetype)
     return -1;
 }
 
+static dmGraphics::TextureImage MakeTextureImage(uint16_t width, uint16_t height, uint8_t max_mipmaps, uint8_t bitspp,
```

Comment:
I think returning structs this large isn't our usual pattern.
It's usually passing in a pointer/reference, or returning an allocated pointer.